### DISCUSSION TITLE
Livewire + Alpine without `@entangle` or `wire:model`

### DISCUSSION THREAD

**Discussion by ju5t** [livewire/livewire#3778] [1. Help] [answered]
We're using a [clipboard](https://github.com/ryangjchandler/alpine-clipboard) (it's really cool!) in some paginated lists. The pagination is done in Livewire. This works, but when you go to page two, there apparently is some DOM-diffing problem. The value from page 1 sticks and is also shown on page 2.

I have been able to replicate this with just a basic Alpine component, so it isn't an issue with the clipboard. A basic component inside a loop would look like this:

```blade
@foreach ($users as $user)
<div wire:key="{{ $user->id }}">
    <span x-data="{ value: '{{ $user->name }}' }">
        <span x-html="value" />
    </span>
   <span>{{ $user->id }}</span>
</div>
@endforeach
```

This doesn't include the clipboard, but a component as simple as this has the same issue. If you would use this in a paginated Livewire component, the user ID would be updated, whereas the name would stick as its inside an Alpine component.

As we don't need two-way communication I am a bit hesitant to `@entangle` it.  

Is there some way to re-init the Alpine component when a user goes to the next page or do I have no other choice and should I `@entangle` it?

**Comment by joshhanley** [comment, marked as answer]
@ju5t try putting another `wire:key` on the same element as the x-data as well, that should tell it to remove old element including Alpine and add a new one.

Hope this helps!

**Reply by ju5t** [reply]
@joshhanley ah, I did that, and it didn't work. Must've done something wrong back then because I've added it now and it starts working! Didn't expect this to be honest.

**Reply by joshhanley** [reply]
@ju5t ah yeah maybe keys not unique or something? Glad it worked though 🙂 

The reason it does this, Alpine doesn't detected changes in `x-data` after it has been loaded. 

So by putting a `wire:key` on the element with the `x-data`, it instructs Livewire to remove that element from the DOM and add a new one, when the key changes.

Then Alpine detects that a new element has been added and that it has `x-data` on it, so it boots itself on that element.

Hope this helps!

**Reply by ju5t** [reply]
@joshhanley thanks for the explanation (as always!)

**Reply by joshhanley** [reply]
No worries 🙂
